### PR TITLE
Toggle Completion

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
**Description of changes**
- Implement Toggle completed endpoint.

**Associated ticket link**
https://trello.com/c/9BOPCXm7

**Screenshot / Screen Recording**

https://github.com/user-attachments/assets/8ee754ac-1957-413d-97a6-c6f147df0d27

### CODE REVIEW

File: src/tasks/tasks.controller.ts [hunk 1]
```diff
@@ -30,4 +32,9 @@ export class TasksController {
   async findAllTasks(): Promise<Task[]> {
     return await this.tasksService.findAllTasks();
   }
+
+  @Patch(":id/complete")
+  async toggleCompletion(@Param("id") id: number): Promise<Task> {
+    return this.tasksService.toggleCompletion(id);
+  }
```
Comment (ishmam-iqbal-sazim) [inline src/tasks/tasks.controller.ts:39]: Do we need this endpoint? Can't we use the update endpoint to achieve this?

Comment (akib-raihan-sazim) [reply src/tasks/tasks.controller.ts:61]: > Do we need this endpoint? Can't we use the update endpoint to achieve this?

In my previous frontend I used a different function for toggling complete button. So I thought that I should separate the function in backend too. To make things clearer for me.

Comment (ishmam-iqbal-sazim) [reply src/tasks/tasks.controller.ts:61]: This is redundant so let's use the update one endpoint from the frontend. You don't need the changes in this PR.

Comment (ishmam-iqbal-sazim) [reply src/tasks/tasks.controller.ts:61]: Yes, you can close this pull request.

**Pull request merged**